src/diffcalc_API/stores/mongo.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Dict, Optional
+
+import numpy as np
+from diffcalc.hkl.calc import HklCalculation
+from diffcalc.hkl.constraints import Constraints
+from diffcalc.ub.calc import UBCalculation
+from pymongo.collection import Collection
+from pymongo.results import DeleteResult
+
+from diffcalc_API.database import database
+from diffcalc_API.errors.definitions import (
+    ALL_RESPONSES,
+    DiffcalcAPIException,
+    ErrorCodes,
+)
+from diffcalc_API.stores.protocol import HklCalcStore
+
+
+class Codes(ErrorCodes):
+    attempting_to_overwrite = 405
+    document_found = 404
+    nothing_to_delete = 404
+
+
+def attempting_to_overwrite(name: str, collection: Collection) -> None:
+    existing_document = collection.find_one({"ubcalc.name": name})
+    if existing_document:
+        message = (
+            f"Document already exists for crystal {name}!"
+            f"\nEither delete via DELETE request to this URL "
+            f"or change the existing properties. "
+        )
+        raise DiffcalcAPIException(
+            status_code=Codes.attempting_to_overwrite, detail=message
+        )
```

Comment:
`error_if_overwrite` would be a more descriptive name as the side effect of this function is not captured by `attempting_to_overwrite`

An alternative structure may be to create a `OverwriteError` which subclasses DiffcalcAPIException and formats an appropriate message, then check if an overwrite is being performed in the `create` method itself

Reply:
I like the sound of the second option. But I think it's best to do that in a separate PR, as my 'pickling.py' file in this folder does the same thing. Noted.

Reply:
+1 to the subclassing idea